test/test_changeset_encoding.cpp

```diff
@@ -22,6 +22,17 @@ Changeset encode_then_parse(const Changeset& changeset)
 }
 } // namespace
 
+TEST(ChangesetEncoding_InternStringsNotDuplicated)
+{
+    sync::ChangesetEncoder encoder;
+
+    encoder.intern_string("Prógram");
+    encoder.intern_string("Program");
+    // Bug #5193 caused "Program" not to be found through the interned strings
+    // although it was just created before.
+    encoder.intern_string("Program");
```

Comment:
I think the original failure here was that when you decoded the changeset produced by this it would fail with a BadChangeset error. Can we round-trip this and check that the changeset encodes and then successfully decodes?

Reply:
There is no validation when the changeset is decoded so the test would pass. I will replace this "test" with a sync test where the server rejects the changeset since that was the original failure.